In [32]:
import pandas as pd
import numpy as np
import tensorflow as tf
import seaborn as sns 
import matplotlib.pyplot as plt
%matplotlib inline 
sns.set(color_codes=True)

Tablica koja prikazuje prosječni score svake osobe za svaku vještinu i koju strategiju je ta osoba odabrala.

In [50]:
df = pd.read_excel("SEVER_2.xlsx")
df.iloc[:,0] = df.iloc[:,0]/10
df.iloc[:,1] = df.iloc[:,1]/10
df.iloc[:,2] = df.iloc[:,2]/10
df

Vjestina aktivnog slusanja  Vjestina umrezavanja  \
0                          4.6                   4.0   
1                          4.8                   4.2   
2                          4.8                   4.8   
3                          4.0                   5.0   
4                          4.0                   4.2   
5                          3.4                   4.4   
6                          4.4                   4.2   
7                          5.0                   4.6   
8                          3.6                   3.2   
9                          3.8                   4.0   
10                         4.0                   4.4   
11                         3.6                   4.4   
12                         4.4                   5.0   
13                         4.8                   4.0   
14                         4.4                   4.6   
15                         3.6                   3.0   
16                         4.4                   4.8   
17                         4.0                   4.4   
18                         4.0                   3.6   
19                         4.6                   3.6   
20                         4.4                   5.0   

    Vjestina utjecaja na druge  Tvrdo pregovaranje  Pozicijsko pregovaranje  \
0                          3.6                   0                        0   
1                          4.4                   0                        1   
2                          4.2                   0                        0   
3                          4.6                   1                        0   
4                          4.4                   0                        0   
5                          4.8                   0                        0   
6                          4.8                   0                        0   
7                          4.2                   0                        0   
8                          3.6                   0                        0   
9                          4.4                   0                        0   
10                         4.8                   0                        0   
11                         4.0                   0                        0   
12                         4.4                   0                        0   
13                         4.0                   0                        0   
14                         4.6                   0                        0   
15                         3.8                   0                        0   
16                         4.4                   0                        0   
17                         3.6                   0                        0   
18                         4.4                   0                        0   
19                         4.8                   0                        0   
20                         5.0                   0                        0   

    Meko pregovaranje  Fiktivno pregovaranje  Principijelno pregovaranje  \
0                   0                      0                           1   
1                   0                      0                           0   
2                   0                      0                           1   
3                   0                      0                           0   
4                   0                      0                           1   
5                   0                      0                           1   
6                   0                      0                           1   
7                   0                      0                           1   
8                   0                      0                           1   
9                   0                      0                           1   
10                  0                      0                           1   
11                  0                      0                           1   
12                  0                      0

In [52]:
# Create an empty list
filter_arr = []
# go through each element in arr
for i in range(0, 21):
  if 'M' == df.iloc[i,9]:
    filter_arr.append(True)
  else:
    filter_arr.append(False)

In [54]:
vas_mladji = df.iloc[:,0][filter_arr].to_numpy()

In [55]:
vas_mladji

array([4.8, 4.8, 4. , 4. , 3.4, 4.4, 4.4])

In [56]:
filter_arr_1 = np.invert(filter_arr)

In [58]:
vas_stariji = df.iloc[:,0][filter_arr_1].to_numpy()

In [59]:
np.mean(vas_mladji)

4.257142857142857

Prosjek za vještinu aktivnog slušanja za osobe mladje od 35 je 4.257...

In [60]:
np.mean(vas_stariji)

4.2

Prosjek za vještinu aktivnog slušanja za osobe koje su starije od 45 je 4.2. 

Mladji u prosjeku imaju malo bolju vjestinu aktivnog slusanja.

In [61]:
vu_mladji = df.iloc[:,1][filter_arr].to_numpy()

In [62]:
vu_stariji = df.iloc[:,1][filter_arr_1].to_numpy()

In [63]:
np.mean(vu_mladji)

4.542857142857143

Prosjek za vjestinu umrezavanja za osobe mladje od 35 je 4.542757..

In [64]:
np.mean(vu_stariji)

4.114285714285715

Prosjek za vjestinu umrezavanja za osobe starije od 45 je 4.114....

Dakle, mladji imaju bolju vjestinu umrezavanja.

In [67]:
vud_mladji = df.iloc[:,2][filter_arr].to_numpy()
vud_stariji = df.iloc[:,2][filter_arr_1].to_numpy()

In [68]:
np.mean(vud_mladji)

4.514285714285714

Prosjek za vjestinu utjecaja na druge za osobe mladje od 35 je 4.5142757...

In [69]:
np.mean(vud_stariji)

4.228571428571429

Prosjek za vjestinu utjecaja na druge za osobe starije od 45 je 4.22757...

Mladji imaju bolje razvijenu vjestinu utjecaja na druge osobe.

## Whitney Mann U Test

Whitney Mann U test ima nul-hipotezu H0: distribucija jedne populacije je jednaka distribuciji druge populacije.

Prvo ćemo usporediti distribuciju vjestine aktivnog slusanja izmedju mladih i starijih.

In [70]:
import scipy
scipy.stats.mannwhitneyu(vas_mladji, vas_stariji)

MannwhitneyuResult(statistic=53.0, pvalue=0.7907225601805892)

Jako velika p-vrijednost (cca 79%), dakle ne odbacujemo H0 hipotezu, odnosno ne odbacujemo hipotezu koja kaze da su distribucije tih dviju populacija jednake.

Odnosno, ne odbacujemo hipotezu koja kaze da su vjestine aktivnog slusanja kod mladjih jednake kao i kod muskaraca.

Sada na isti nacin ispitamo za vjestinu umrezavanja

H0: vjestine umrezavanja za mladje osobe imaju jednaku distribuciju kao i vjestine umrezavanja starijih osoba

In [72]:
import scipy
scipy.stats.mannwhitneyu(vu_mladji, vu_stariji)

MannwhitneyuResult(statistic=69.0, pvalue=0.14244517571439017)

p-vrijednost je 14.2445%, dakle ne odbacujemo H0 hipotezu.

I na isti nacin ispitamo i za trecu vjestinu.

In [73]:
import scipy
scipy.stats.mannwhitneyu(vud_mladji, vud_stariji)

MannwhitneyuResult(statistic=66.5, pvalue=0.19696367173141816)

P-vrijednost je 19.69636717..%, dakle ne odbacujemo H0.